Sources/Filters/Texture/TextureMapToPlane/index.js

```diff
@@ -87,19 +87,23 @@ function vtkTextureMapToPlane(publicAPI, model) {
     const c1 = [m[0], m[1], m[2]];
     const c2 = [m[3], m[4], m[5]];
     const c3 = [m[6], m[7], m[8]];
-    const matrix = [c1, c2, c3];
+    const matrix = m;
     const det = vtkMath.determinant3x3(matrix);
     if (det <= VTK_TOLERANCE) {
       return;
     }
 
-    matrix[0] = v;
-    matrix[1] = c2;
-    matrix[2] = c3;
+    for (i = 0; i < 3; i++) {
+      matrix[i] = v[i];
+      matrix[i + 3] = c2[i];
+      matrix[i + 6] = c3[i];
+    }
```

Comment:
for performance reasons, I would simply do `matrix = [v[0], v[1], v[2], c2[0], ...]` instead of a for loop.